### PR TITLE
Introduce instrumentation report utilities

### PR DESCRIPTION
**Pull request by AlexJones0**
This PR is the eighth of a series of PRs to introduce instrumentation reporting to DVSim.

This PR adds some helper utilities that will be used throughout the various instrumentation visualizations. These are not used yet in this PR - in an effort to split the changes up into multiple PRs and make them easier to review. There are a variety of helpers across the commits - some for time string formatting, some for retrieving concrete timing information from the report data, and some for rendering Plotly figures and deriving consistent colour mappings across visualizations.

(note on the time string formatting specifically: `datetime` does let you get some format like "Xh Ym Zs", but it doesn't let you omit leading zero values which is nice for presentation in the reports. We could hack on top and strip leading string prefixes but it's simple enough to just re-implement that I decided to do that here. We also could pull in some third-party deps, but that seems unnecessary for just this functionality).

See the commit messages for more information.

### CODE REVIEW

File: src/dvsim/utils/time.py [hunk 1]
```diff
@@ -12,3 +12,44 @@
 
 # Timestamp format for Hours:Minutes:Seconds display as used by the scheduler
 TS_HMS_FORMAT = "%H:%M:%S"
+
+
+def format_time_as_hms(seconds: float, *, decimals: int = 2, omit_zero: bool = False) -> str:
```
Comment (machshev) [inline src/dvsim/utils/time.py:17]: Looks good but I'd probably add some parameterize unit tests for this to catch corner cases or in case someone updates the implementation and breaks it?

File: src/dvsim/instrumentation/report/base.py [hunk 2]
```diff
@@ -130,3 +150,90 @@ def render_html_report(
             (outdir / plotly_js_path).write_text(artifacts[plotly_js_path])
 
     return artifacts
+
+
+def render_plotly_figure(fig: go.Figure, *args: list[Any], **kwargs: dict[str, Any]) -> str:
```
Comment (machshev) [inline src/dvsim/instrumentation/report/base.py:155]: `go.Figure` ;)

**Review by machshev — APPROVED** [review]
Thanks @AlexJones0

File: src/dvsim/instrumentation/report/base.py (hunk 2 above)
Comment (AlexJones0) [reply src/dvsim/instrumentation/report/base.py:155]: Several times I've found this amusing whilst working on this code :)

File: src/dvsim/utils/time.py (hunk 1 above)
Comment (AlexJones0) [reply src/dvsim/utils/time.py:17]: Great point, I didn't think of this at all. I'll add it now.

Comment (AlexJones0) [reply src/dvsim/utils/time.py:17]: Added.
